docs/docs/references/users/README.md

```diff
@@ -0,0 +1,97 @@
+# 👨‍👩‍👧‍👦 Users
+
+The _users_ are the main entities of the identity service.
+We will describe the user-related concepts and details in the following.
+
+## Profile
+
+Each user has a profile containing the user information, more than the name and contact details.
+
+A user profile consists of 3 sections:
+
+- **Basic data**: contains basic info in pre-defined attributes, such as user id, username, email, phone number, and when the user last signed in.
+- **Custom data**: contains additional info in customizable attributes, such as user-preferred color and language.
+- **Identities**: contains the user info retrieved from social sign-in, such as Facebook, GitHub, and WeChat.
+
+See [`packages/schemas/tables/users.sql`](https://github.com/logto-io/logto/blob/master/packages/schemas/tables/users.sql) for the latest version of the user table definition in the relational database.
+
+### All attributes
+
+The following attributes (except password-related) are visible on the user profile.
+
+:::note The meanings of some following table columns
+
+- **Unique**: Ensures the _uniqueness_ of the values entered into an attribute of a DB table.
+- **Required**: Ensures that the values entered an attribute of a DB table can NOT be NULL.
+- **Editable**: Admin users can only edit the _editable_ attributes using Admin Console or Management API.
+- **Searchable**: Admin users can only search the keyword on the _searchable_ attributes using Admin Console or Management API.
+
+:::
+
+<br/>
+
+| Name                        | Type         | Description                                   | Unique | Required | Editable | Searchable |
+| --------------------------- | ------------ | --------------------------------------------- | ------ | -------- | -------- | ---------- |
+| id                          | text         | Unique identifier, auto-generate on creating  | ✅     | ✅       | ❌       | ❌         |
+| username                    | text         | Username for sign-in                          | ✅     | ❌       | ❌       | ✅         |
+| primary_email               | text         | Primary email                                 | ✅     | ❌       | ✅       | ✅         |
+| primary_phone               | text         | Primary phone number                          | ✅     | ❌       | ✅       | ✅         |
+| name                        | text         | Full name                                     | ❌     | ❌       | ✅       | ✅         |
+| avatar                      | text         | URL pointing to user's avatar image           | ❌     | ❌       | ✅       | ❌         |
+| role_names                  | string array | List of roles                                 | ❌     | ✅       | ❌       | ❌         |
```

Comment:
Maybe add some comment on this special attribute, and they are actually editable through management API.

Reply:
will describe it in the next section `basic data`, `custom data`, `identities` to avoid this table becoming too complicated.

will also link the attribute to the description in the next section.